### DISCUSSION TITLE
Flash via J1 connector

### DISCUSSION THREAD

**Discussion by stonewalljones** [revk/xESP32-Faikout#207] [Q&A]
I order some DIY board from PCBway and I'm just getting around to flashing them.
I have a 5v/3.3v USB to TTL Serial Adapter. Is it possible to flash the board via the J1 connector at the bottom or will I need to solder either the usb pads or pads at the top of the board?

**Comment by revk** [comment]
What does the board look like?

**Reply by stonewalljones** [reply]
Err the Serial Adapter or the boards I DIY'd?
The serial adapter is this one: https://www.amazon.com/dp/B07BBPX8B8?psc=1&ref=ppx_yo2ov_dt_b_product_details
and I'll upload some picture of the boards momentarily

**Reply by stonewalljones** [reply]
![PXL_20240128_030221154.RAW-01.COVER.jpg](https://github.com/revk/ESP32-Faikin/assets/6126541/5d379f55-cd18-483a-82a5-581188edb520)

![PXL_20240128_030208816.RAW-01.COVER.jpg](https://github.com/revk/ESP32-Faikin/assets/6126541/03f6ab0a-814e-494b-8596-d192c0c0b346)

**Reply by revk** [reply]
Ok... you need to select 3.3V in the adaptor.

The pads marked T R + 0 ⏚

⏚ and + are the power.

T is Tx from Faikin
R is Rx to Faikin
0 will probably want to connect to DTR

But surely the board was already flashed with code? Does it not work when connected?

**Reply by stonewalljones** [reply]
These were DIY ones I had order not from you. My tty adaptor didn't have enough power but I managed to sacrifice a USB cable and flash that way.

**Reply by stonewalljones** [reply]
However it's still not booting. the esp32 is properly getting 3.3v but no led activity or wifi networks
this is my flash command anything look out of place?

esptool -p /dev/ttyACM0 write_flash 0x1000 Faikin-S3-MINI-N4-R2-bootloader.bin 0x8000 partition-table.bin 0xd000 ota_data_initial.bin 0x10000 Faikin-S3-MINI-N4-R2.bin

**Reply by revk** [reply]
Does it boot when on the serial lead or USB connection?

**Reply by stonewalljones** [reply]
I'm able to interact with it via esptool on USB but it doesn't light up the LED or start any wifi networks

**Reply by revk** [reply]
That sounds like you have not flashed right, does it kit say anything useful in esptool monitor?

**Comment by beaulif** [comment]
Hello, I also have problems with the flash of the ESP32-S3-MINI-1-N4R2. I order some DIY board from JLCPCB the version with the 2 protection diodes. I used flash_download_tool_3.9.5 (see image) with FTDI. I set gnd to gpio 0 to put it in programming mode. I don't have AP and LED indication. Do you have a possible solution?
Thanks for your help!
![Capture](https://github.com/revk/ESP32-Faikin/assets/7018127/4cdc6d8e-6951-4b48-bd99-15684c166334)

**Reply by revk** [reply]
Someone may be able to help, but that is not a tool I have used. I just use the ESP IDF (command line) stuff.

**Reply by beaulif** [reply]
Hi,  I still don't have AP after the flash. Do you see my problem?
Thanks a lot for your help.

I installed esp-idf-v5.1.2 and esptool.py v4.7.0. When I use this command:

esptool.py -p com3 -b 115200 --before default_reset --after hard_reset --chip esp32-s3 write_flash --flash_mode dio --flash_size detect --flash_freq 40m 0x1000 Faikin-S3-MINI-N4-R2-bootloader.bin 0x8000 partition-table.bin 0xd000 ota_data_initial.bin 0x10000 Faikin-S3-MINI-N4-R2.bin

resulted:
PS C:\Espressif\faikin> esptool.py -p com3 -b 115200 --before default_reset --after hard_reset --chip esp32-s3 write_flash --flash_mode dio --flash_size detect --flash_freq 40m 0x1000 Faikin-S3-MINI -N4-R2-bootloader.bin 0x8000 partition-table.bin 0xd000 ota_data_initial.bin 0x10000 Faikin-S3-MINI-N4-R2.bin
esptool.py v4.7.0
Serial port com3
Connecting...
Failed to get PID of a device on com3, using standard reset sequence.
.
Chip is ESP32-S3 (QFN56) (revision v0.1)
Features: WiFi, BLE, Embedded Flash 4MB (XMC), Embedded PSRAM 2MB (AP_3v3)
Crystal is 40MHz
MAC: dc:54:75:ef:52:9c
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 4MB
Flash will be erased from 0x00001000 to 0x00004fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x0000d000 to 0x0000eff...
Flash will be erased from 0x00010000 to 0x00150fff...
Compressed 13712 bytes to 9772...
Wrote 13712 bytes (9772 compressed) at 0x00001000 in 1.1 seconds (effective 104.3 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 194...
Wrote 3072 bytes (194 compressed) at 0x00008000 in 0.1 seconds (effective 393.0 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 31...
Wrote 8192 bytes (31 compressed) at 0x0000d000 in 0.1 seconds (effective 695.2 kbit/s)...
Hash of data verified.
Compressed 1314816 bytes to 861492...
Wrote 1314816 bytes (861492 compressed) at 0x00010000 in 76.3 seconds (effective 137.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

**Comment by revk** [comment]
What does *monitor* show?

**Reply by beaulif** [reply]
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0x178c
ets_loader.c 78

**Reply by revk** [reply]
Well that is constantly rebooting, so not flashed right!

You are flashing the right build for the chip?

**Reply by beaulif** [reply]
i use file from github.com/revk/ESP32-Faikin/tree/main/ESP32/release
partition-table.bin
ota_data_initial.bin
Faikin-S3-MINI-N4-R2-bootloader.bin
Faikin-S3-MINI-N4-R2.bin

**Reply by revk** [reply]
And your board definitely has an ESP32-S3-MINI-N4-R2, yes?

**Reply by beaulif** [reply]
yes.  i flash it whit example blink and de led flash.

**Reply by revk** [reply]
Do you have a picture of the board? (close up)

**Reply by beaulif** [reply]
![espfaikin](https://github.com/revk/ESP32-Faikin/assets/7018127/92a021a5-08d6-4f72-a0f5-797b130b2e20)

**Reply by revk** [reply]
That looks right - so at a loss what you are doing wrong. You could instal the EDP IDF and build from scratch maybe.

I'm sure others have flashed from those binaries. Maybe someone else has a suggestion. The addresses looked right to me.

**Comment by sam-m7** [comment]
Hi @stonewalljones and @beaulif, 
I think the problem is the bootloader address. For the S3 versions 0x0 needs to be used. 0x1000 was/is only valid for the S1 versions.

**Reply by beaulif** [reply]
you are right @sam-m7.  👍  🥇 It works with release.bin

![image](https://github.com/revk/ESP32-Faikin/assets/7018127/ef24c5fb-c636-40c5-a690-72de5d1d6478)
